src/datagen/groovy/techreborn/datagen/recipes/TechRebornRecipesProvider.groovy

```diff
@@ -50,13 +50,16 @@ abstract class TechRebornRecipesProvider extends FabricRecipesProvider {
     abstract void generateRecipes()
 
     static Ingredient createIngredient(def input) {
-        if (input instanceof ItemConvertible) {
+        if (input instanceof Ingredient) {
+			return input
+		}
+		if (input instanceof ItemConvertible) {
```

Comment:
Add TagConvertible before ItemConvertible? Should let you remove all of the `.asTag()` calls unless im missing something?

Reply:
I'm unsure if we want that. There might be instances where it is better to call the TR id instead of the common tag. So using the tag as of now requires a conscious decision

Reply:
Ok, thats fine 👍 can always change it later.